File: old/actions/install_glusterfs.py
"""
Install GlusterFS server action
"""
import logging
from .base import Action
from cli import Apt

logger = logging.getLogger(__name__)


class InstallGlusterfsAction(Action):
    """Install GlusterFS server packages"""
    description = "glusterfs server installation"

    def execute(self) -> bool:
        """Install GlusterFS server and client packages"""
        logger.info("Installing GlusterFS server and client packages...")
        
        # Update apt first
        update_cmd = Apt.update_cmd()
        update_output, exit_code = self.pct_service.execute(
            str(self.container_id), update_cmd, timeout=600
        )
        if exit_code is not None and exit_code != 0:
            logger.error("Failed to update apt: %s", update_output)
            return False
        
        # Install GlusterFS packages
        install_cmd = Apt.install_cmd(["glusterfs-server", "glusterfs-client"])
        install_output, exit_code = self.pct_service.execute(
            str(self.container_id), install_cmd, timeout=300
        )
        if exit_code is not None and exit_code != 0:
            logger.error("Failed to install GlusterFS: %s", install_output)
            return False
        
        # Enable and start glusterd service
        enable_cmd = "systemctl enable glusterd && systemctl start glusterd"
        enable_output, exit_code = self.pct_service.execute(
            str(self.container_id), enable_cmd, timeout=30
        )
        if exit_code is not None and exit_code != 0:
            logger.error("Failed to start glusterd: %s", enable_output)
            return False
        
        logger.info("GlusterFS server installed and started successfully")
        return True

File: src/actions/install_glusterfs.go
package actions

import (
	"strconv"
	"enva/cli"
	"enva/libs"
	"enva/services"
)

// InstallGlusterfsAction installs GlusterFS server packages
type InstallGlusterfsAction struct {
	*BaseAction
}

func NewInstallGlusterfsAction(sshService *services.SSHService, aptService *services.APTService, pctService *services.PCTService, containerID *string, cfg *libs.LabConfig, containerCfg *libs.ContainerConfig) Action {
	return &InstallGlusterfsAction{
		BaseAction: &BaseAction{
			SSHService:   sshService,
			APTService:   aptService,
			PCTService:   pctService,
			ContainerID: containerID,
			Cfg:          cfg,
			ContainerCfg: containerCfg,
		},
	}
}

func (a *InstallGlusterfsAction) Description() string {
	return "glusterfs server installation"
}

func (a *InstallGlusterfsAction) Execute() bool {
	containerIDInt, err := strconv.Atoi(*a.ContainerID)
	if err != nil {
		libs.GetLogger("install_glusterfs").Printf("Invalid container ID: %s", *a.ContainerID)
		return false
	}
	libs.GetLogger("install_glusterfs").Printf("Installing GlusterFS server and client packages...")
	updateCmd := cli.NewApt().Update()
	updateOutput, exitCode := a.PCTService.Execute(containerIDInt, updateCmd, libs.IntPtr(600))
	if exitCode != nil && *exitCode != 0 {
		libs.GetLogger("install_glusterfs").Printf("Failed to update apt: %s", updateOutput)
		return false
	}
	installCmd := cli.NewApt().Install([]string{"glusterfs-server", "glusterfs-client"})
	installOutput, exitCode := a.PCTService.Execute(containerIDInt, installCmd, libs.IntPtr(300))
	if exitCode != nil && *exitCode != 0 {
		libs.GetLogger("install_glusterfs").Printf("Failed to install GlusterFS: %s", installOutput)
		return false
	}
	enableCmd := "systemctl enable glusterd && systemctl start glusterd"
	enableOutput, exitCode := a.PCTService.Execute(containerIDInt, enableCmd, libs.IntPtr(30))
	if exitCode != nil && *exitCode != 0 {
		libs.GetLogger("install_glusterfs").Printf("Failed to start glusterd: %s", enableOutput)
		return false
	}
	libs.GetLogger("install_glusterfs").Printf("GlusterFS server installed and started successfully")
	return true
}


